concept/type/ThingType.java

```diff
@@ -35,15 +37,17 @@ public interface ThingType extends Type {
     @Override
     Forwardable<? extends ThingType, Order.Asc> getSupertypes();
 
+    Forwardable<? extends ThingTypeImpl, Order.Asc> getSupertypesWithThing();
+
```

Comment:
`getSupertypes()` now only goes up to the root entity/relation/attribute type, as it now requires that supertypes are of the same class as the original type. However, since some behaviour still relies on getting the root thing type in the list of supertypes, this band-aid is required until the `thing` type is fully removed.